src/nvim/misc1.c

```diff
@@ -2927,12 +2930,11 @@ char_u *vim_getenv(char_u *name, int *mustfree)
     return p;
   }
 
-  vimruntime = (STRCMP(name, "VIMRUNTIME") == 0);
```

Comment:
For NVIMRUNTIME, at least, I think we should do a case-sensitive comparison. Otherwise it's just more surface area to cover in tests, troubleshooting, and documentation.